interfaces/builtin/basedeclaration.go

```diff
@@ -217,6 +217,11 @@ slots:
       slot-snap-type:
         - core
     deny-auto-connection: true
+  download:
+    allow-installation:
+      slot-snap-type:
+        - app
+    deny-auto-connection: true
```

Comment:
My understanding of UDM is that its DBus is considered completely safe for anyone to use. As such, this should drop 'deny-auto-connection: true'. Because it is is slot implementation, it should have 'deny-connection: true' to be in line with our current implementation.

Note that the use of 'deny-connection' is actively being discussed and we may change this. When developing this interface, please also note https://bugs.launchpad.net/snappy/+bug/1640874.